python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -2660,6 +2673,17 @@ def get_asset_check_executions(
             for row in rows
         ]
 
+        # filter to only include records created after the last wipe timestamp. It's ok to do this filtering
```

Comment:
the ideal would be to do this filtering in the query, but that runs in to the issue that we have a datetime column instead of timestamp, and there's (surely solvable) issues where the datetime is wrong based on the timezone of the postgres instance. We avoid this with events by having a timestamp column.

I don't think we're going to support any ordering other than timestamp in the medium future though, so I think filtering here is fine

Reply:
So, is it too late to add a timestamp col?   

In cloud, we're also just deleting rows from the new asset tables for the wiped asset key, while leaving the event log intact.  This generally means we don't need to apply any timestamp filter.

Reply:
Two great points.

We could delete check results on wipe. The one implication is that I think we'd need to support writing check results without an initial PLANNED row. That's not the end of the world, but that invariant has been useful for catching bugs where the planned event isn't getting written.

We could also do a new migration before 1.5 to support wipes via timestamp.

@prha  Just in terms of queries, would you prefer deletes (for both oss and cloud)?

Reply:
Mmm, I'm not sure... it makes the wipe action more expensive for sure.  Without the right indexes (by asset_key) the wipe mutation would probably time out.  But I think it might make sense, otherwise we're keeping around rows that will never be read again (will always get filtered out).  I think if the read is cheap (there exists an index with timestamp in it), it might not make that much difference.

Just for data scalability, we'd probably opt to delete in cloud.  Also helps that wipes are relatively rare. 

Another option is to do it by timestamp, but have some periodic process that deletes wiped rows.

Reply:
I'm going to go with deletes. I'll post another diff